build/ci/phase-template.yml

```diff
@@ -36,5 +36,17 @@ phases:
         testRunTitle: Machinelearning_Tests_$(_phaseName)_$(_configuration)_$(Build.BuildNumber)
         configuration: $(_configuration)
         mergeTestResults: true
+    - task: CopyFiles@2
+      displayName: Stage build logs
+      inputs:
+        sourceFolder: $(Build.SourcesDirectory)
+        contents: '?(msbuild.*|binclash.log|init-tools.log|bin/**/TestOutput/**/*)'
+        targetFolder: $(Build.ArtifactStagingDirectory)
+    - task: PublishBuildArtifacts@1
+      displayName: Publish build and test logs
+      inputs:
+        pathToPublish: $(Build.ArtifactStagingDirectory)
+        artifactName: ${{ parameters.name }}$(_configuration)
```

Comment:
should we also add architecture in the naming? So that they don't clash? For example in the Windows builds?

Reply:
The parameters.name here is passed in as "Windows_x64" and "Windows_x86", so we have that covered.  There is actually a filename length problem here.  I added 14 characters at it pushed it over the limit, so I've got to figure out how to make unique names within that restriction (and hope we don't make any tests with names longer than `CommandTrainingBinaryFieldAwareFactorizationMachineWithValidationAndInitialization`)

Reply:
True, got confused at 1 point, thought that the architecture was expanded through a matrix, but no it is by referencing the template with different architectures. 

🤣 hopefully there are no tests with a longer name than that.